src/Unity.Mathematics/random.cs

```diff
@@ -27,6 +27,50 @@ public Random(uint seed)
             NextState();
         }
 
+        /// <summary>
+        /// Constructs a <see cref="Random"/> instance with a seed that gets hashed.  The seed must be non-zero.
+        /// </summary>
+        /// <remarks>
+        /// Use this function when you expect to create several Random instances from consecutive or
+        /// very similar seeds.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// for (uint i = 0; i &lt; 4096; ++i)
+        /// {
+        ///     // Add 1 to i to avoid using zero.
+        ///     Random rand = Random.CreateFromHashedSeed(i + 1u);
+        ///
+        ///     // Random numbers drawn from loop iteration j should be very different
+        ///     // from every other loop iteration k.
+        ///     rand.NextUInt();
+        /// }
+        /// </code>
+        /// </example>
+        /// <param name="seed_to_hash">A seed that will be hashed for Random creation.  Must be non-zero.</param>
+        /// <returns><see cref="Random"/> created from hashed seed.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Random CreateFromHashedSeed(uint seed_to_hash)
```

Comment:
I think the name of this function is misleading. `CreateFromHashedSeed` sounds like it takes a hashed seed as its input. Maybe something like `CreateFromIndex`? Not sure.

Reply:
I do like the name `CreateFromIndex` better, but I don't know how to use that name and still achieve clear expectations on what inputs are valid.

Concretely, `CreateFromIndex` I think would make sense to accept zero as valid input but to make that happen, I need to pick some other index which is invalid for use.  Which index do I pick?  If I pick one that the user actually needs, they need to be able to do something about it to work around the limitation.  I could document it but I'm not a huge fan of that approach.

Because I don't know how to answer that question in every case, it seemed simpler to follow the exact same restrictions as the constructor does (must be non-zero).

An unintended, but nice benefit of keeping the same input requirements is that it's trivial to switch between using the constructor and this new function.

Reply:
I agree that CreateFromIndex is a good name, and I agree that 0 should be a valid input (because it's a very common index!). I'm OK if the invalid input becomes the pattern of all-1 bits, or something. Since it's a negative index anyway.

Reply:
Making this `CreateFromIndex` and allowing 0 to be a valid input.  uint.MaxValue will be invalid.